src/imap/idle.rs

```diff
@@ -63,42 +60,46 @@ impl Session {
         handle.as_mut().set_read_timeout(None);
         let (idle_wait, interrupt) = handle.wait_with_timeout(IDLE_TIMEOUT);
 
-        enum Event {
-            IdleResponse(IdleResponse),
-            Interrupt,
-        }
-
         info!(
             context,
             "IDLE entering wait-on-remote state in folder {folder:?}."
         );
-        let fut = idle_wait.map(|ev| ev.map(Event::IdleResponse)).race(async {
-            idle_interrupt_receiver.recv().await.ok();
 
-            // cancel imap idle connection properly
-            drop(interrupt);
+        // Spawn a task to relay interrupts from `idle_interrupt_receiver`
+        // into interruptions of IDLE.
+        let interrupt_relay = {
+            let context = context.clone();
+            let folder = folder.to_string();
+
+            tokio::spawn(async move {
+                idle_interrupt_receiver.recv().await.ok();
 
-            Ok(Event::Interrupt)
-        });
+                info!(context, "{folder:?}: Received interrupt, stopping IDLE.");
 
-        match fut.await {
-            Ok(Event::IdleResponse(IdleResponse::NewData(x))) => {
+                // cancel imap idle connection properly
+                drop(interrupt);
+            })
```

Comment:
```suggestion
                // Drop `interrupt` in order to cancel the imap idle connection
                drop(interrupt);
            })
```

or:

```suggestion
                // Cancel the imap idle connection
                drop(interrupt);
            })
```

I found the "properly" confusing when reading, since it's not about doing it properly but doing it at all (which implies that we do it properly, since in general, we're doing most things properly)